packages/vm/src/evm/precompiles/02-sha256.ts

```diff
@@ -8,12 +8,11 @@ export default function (opts: PrecompileInput): ExecResult {
 
   const data = opts.data
 
-  const gasUsed = new BN(opts._common.param('gasPrices', 'sha256'))
-  gasUsed.iadd(
-    new BN(opts._common.param('gasPrices', 'sha256Word')).imuln(Math.ceil(data.length / 32))
-  )
+  let gasUsed = BigInt(opts._common.param('gasPrices', 'sha256'))
```

Comment:
Not for this PR and a bit of a side note: but I guess these continued Common parameter reads (+ re-conversions) to BigInt all over again will really cost us a lot of performance (think this was in discussion before in some way? Also mentioned (in a somewhat different context) in Patricios performance low-hanging-fruits issue #775.

I guess a complete refactor (of the Common JSON files) will likely overload us (and others who are already using this format), but what we could (and should likely) really do is caching these values - as BigInt - in Common on first read and then always return these values from the cache (and: generally return BigInt values on `param()` calls and similar.

Does this make sense?

Maybe also something for @emersonmacro being the caching expert. 😋

Reply:
nice idea, just thinking off the top of my head, maybe we add an optional parameter to `param()` with `asBigInt = false`, then the signature could be something like:

```typescript
param(topic: string, name: string, asBigInt: true): bigint
param(topic: string, name: string, asBigInt = false): number
```

another option would be to add a new method e.g. `paramAsBigInt(topic: string, name: string): bigint`

would be great to see if an internal cache would help with speedup too

Reply:
Do you think this is necessary to split this up? 🤔

Don't we need most of this as BigInt anyhow and should therefore go full-bigint with the function? Not sure, might need some confirmation first.

Another note: I guess I gave the wrong context for the caching suggestion from Patricio. If I recall correctly he rather meant caching in the VM context (e.g., but also mainly) to avoid this repeated calling into the `Common` functions.

Nevertheless: a number -> BigInt cache in Common itself might independently also worth to investigate.

Reply:
yes we could just return bigint, i think in some cases a number is fine so didn't want to do extra work and then have to additionally call `toNumber()`, but maybe we can do a rough count and see how often bigint vs number is used, and if number is only a few non-essential cases we can go for bigint only.

Reply:
Have done some rough search on this (all monorepo, not only VM), this is extremely strongly on the BigInt/BN usage side, so I would roughly say something like 10:1.

(I would be calling occurences with these *n() functions from BN in - so like `addn()` - that makes sense, right?)

Reply:
> (I would be calling occurences with these *n() functions from BN in - so like `addn()` - that makes sense, right?)

not really sure what you mean. the search would be for `common.param` wrapped with `new BN()`

Reply:
I am also calling occurrences in where a value is taken for an input for e.g. the `addn()` method, since after the conversion to BigInt this kind of arithmetic would be implemented using two BigInt values. So then it would also make sense to have a BigInt return value for these kind of situations.